armoryengine/Transaction.py

```diff
@@ -430,6 +431,8 @@ def TxInExtractAddrStrIfAvail(txinObj):
       return hash160_to_addrStr( hash160(lastPush) )
    elif scrType == CPP_TXIN_SPENDP2SH:
       return binScript_to_p2shAddrStr(lastPush)
+   elif scrType == CPP_TXIN_P2WPKH_P2SH or scrType == CPP_TXIN_P2WSH_P2SH:
+      return binScript_to_p2shAddrStr(hash160(rawScript))
    else:
```

Comment:
this could be accomodated to spit out the address for wp2pkh and wp2sh. it's expected we'll have the witness data available in the txinObj.